### PR TITLE
JW-1074 : Don't use binary masks 'valid' and 'modified'

### PR DESCRIPTION
**Pull request by kaweue**

### CODE REVIEW

**Comment by kaweue** [comment]
Binary masks 'valid' and 'modified' limited max number of fields to 64. In master code base there is a type "Tag" which has 82 members.

**Comment by kaweue** [comment]
Retest this please

File: golang_api.py [hunk 1]
```diff
@@ -397,7 +404,7 @@ def _GenerateReferenceRead(self, ident, ref, suffix, file):
         decl = """
 func (obj *%(typecamel)s) read%(fieldcamel)s%(methodsuffix)s() error {
         if !obj.IsTransient() &&
-                (obj.valid & %(typeid)s_%(fieldid)s%(suffix)s == 0) {
+                (!obj.valid[%(typeid)s_%(fieldid)s%(suffix)s]) {
```
Comment (kaweue) [inline golang_api.py]: These parenthesis are not needed

File: golang_api.py [hunk 2]
```diff
@@ -488,18 +495,18 @@ def _GenerateReferenceModifiers(self, ident, link_info, file):
...
         }
-        obj.modified |= %(typeid)s_%(fieldid)s_refs
+        obj.modified[%(typeid)s_%(fieldid)s_refs] = true
         return nil
 }
 
 func (obj *%(typecamel)s) Clear%(fieldcamel)s() {
-        if (obj.valid & %(typeid)s_%(fieldid)s_refs != 0) &&
-           (obj.modified & %(typeid)s_%(fieldid)s_refs == 0) {
+        if (obj.valid[%(typeid)s_%(fieldid)s_refs]) &&
```
Comment (kaweue) [inline golang_api.py]: These parenthesis are not needed

File: golang_api.py [hunk 3]
```diff
@@ -756,7 +765,7 @@ def _GenerateUpdateReferences(self, ident, file):
                 continue
             link_to = ident.getLinkTo(link_info)
             decl = """
-        if (obj.modified & %(typeid)s_%(fieldid)s_refs != 0) &&
+        if (obj.modified[%(typeid)s_%(fieldid)s_refs]) &&
```
Comment (kaweue) [inline golang_api.py]: These parenthesis are not needed

File: golang_api.py [hunk 4]
```diff
@@ -325,13 +331,14 @@ def _GenerateGenericMethods(self, ident, file):
...
 func (obj *%(camel)s) UpdateDone() {
-        obj.modified = 0
+        for i := range obj.modified { obj.modified[i] = false }
         obj.baseMap = nil
 }
 
 """ \
         % {"camel": ident.getCppName(),
            "typename": ident.getName(),
+           'typeid': ident.getCIdentifierName(),
```
Comment (mgreg0r) [inline golang_api.py]: It's not necessary

File: golang_api.py [hunk 5]
```diff
@@ -591,8 +598,10 @@ def _GenerateUnmarshalJSON(self, ident, file):
         if err != nil {
                 return err
         }
+
         for key, value := range m {
-                switch key {""" % {'camel': ident.getCppName()}
+                switch key {""" % {'camel': ident.getCppName(),
+                                   'typeid' : ident.getCIdentifierName()}
```
Comment (mgreg0r) [inline golang_api.py]: It's not necessary

**Review by mgreg0r — DISMISSED** [review]

**Review by mclapinski — APPROVED** [review]
great job :1st_place_medal:

**Pull request merged**